lib/check_data_points.rb

```diff
@@ -0,0 +1,156 @@
+  class << self
+    SEARCH_RESULT_TYPES = ['relevant_search_result_requests', 'irrelevant_search_result_requests', 'timeout_search_requests']
+    GRANULARITY_VALUES = ['year', 'quarter', 'month', 'week', 'day']
+
+    # Number of tipline messages
+    def tipline_messages(team_id, start_date, end_date, granularity = nil)
+      start_date, end_date = parse_start_end_dates(start_date, end_date)
+      query = TiplineMessage.where(team_id: team_id, created_at: start_date..end_date)
+      query_based_on_granularity(query, granularity)
+    end
+
+    # Number of tipline requests
+    def tipline_requests(team_id, start_date, end_date, granularity = nil)
+      start_date, end_date = parse_start_end_dates(start_date, end_date)
+      query = TiplineRequest.where(team_id: team_id, created_at: start_date..end_date)
+      query_based_on_granularity(query, granularity)
+    end
+
+    # Number of tipline requests grouped by type of search result
+    def tipline_requests_by_search_type(team_id, start_date, end_date)
+      start_date, end_date = parse_start_end_dates(start_date, end_date)
+      TiplineRequest.where(
+        team_id: team_id,
+        smooch_request_type: SEARCH_RESULT_TYPES,
+        created_at: start_date..end_date,
+      ).group('smooch_request_type').count
+    end
+
+    # Number of Subscribers
+    def tipline_subscriptions(team_id, start_date, end_date, granularity = nil)
+      start_date, end_date = parse_start_end_dates(start_date, end_date)
+      query = TiplineSubscription.where(team_id: team_id, created_at: start_date..end_date)
+      query_based_on_granularity(query, granularity)
+    end
+
+    # Number of Newsletters sent
+    def newsletters_sent(team_id, start_date, end_date, granularity = nil)
+      start_date, end_date = parse_start_end_dates(start_date, end_date)
+      query = TiplineNewsletterDelivery
+      .joins(:tipline_newsletter)
+      .where('tipline_newsletters.team_id': team_id)
+      .where(created_at: start_date..end_date)
+      query_based_on_granularity(query, granularity, 'newsletter')
+    end
+
+    # Number of Media received, by type
+    def media_received_by_type(team_id, start_date, end_date)
+      bot = BotUser.smooch_user
+      start_date, end_date = parse_start_end_dates(start_date, end_date)
+      ProjectMedia.where(team_id: team_id, user_id: bot.id, created_at: start_date..end_date)
+      .joins(:media).group('medias.type').count
+    end
+
+    # Top clusters
+    def top_clusters(team_id, start_date, end_date, limit = 5)
+      elastic_search_top_items(team_id, start_date, end_date, limit)
+    end
+
+    # Top media tags
+    def top_media_tags(team_id, start_date, end_date, limit = 5)
+      elastic_search_top_items(team_id, start_date, end_date, limit, true)
+    end
+
+    # Articles sent
+    def articles_sent(team_id, start_date, end_date)
+      start_date, end_date = parse_start_end_dates(start_date, end_date)
+      c1 = TiplineRequest.where(
```

Comment:
Suggestion: For clarity, please rename this variable and/or add a comment that explains that "this is the number of articles sent as search results".